lib/push-jenkins-update.js

```diff
@@ -29,13 +29,10 @@ function pushStarted (options, build, cb) {
     }
   }
 
-  findLatestCommitInPr(optsWithPr, (err, latestCommit) => {
-    if (err) {
-      logger.error(err, 'Got error when retrieving GitHub commits for PR')
-      cb(err)
-      return
-    }
-
+  findLatestCommitInPr(optsWithPr).catch(err => {
+    logger.error(err, 'Got error when retrieving GitHub commits for PR')
+    return cb(err)
+  }).then(latestCommit => {
```

Comment:
I think chaining like this will still call the `then` callback even in case of an error. To avoid that, you can use `findLatestCommitInPr(optsWithPr).then(onSuccess, onError)`

Reply:
Humm, interesting, I guess that makes sense. Will test tomorrow and update the code if that's the case.